mitodl/open-discussions
# Add "course feature tags" to index for ocw and remove some obsolete code

Depends on a [new version of ocw-data-parser](https://github.com/mitodl/ocw-data-parser/pull/128) that should be integrated into open-discussions, and a fresh ocw import run that saves the correct parsed json.  PR https://github.com/mitodl/open-discussions/pull/3321

#### Pre-Flight checklist

#### What are the relevant tickets?
Closes #3315 

#### What's this PR do?
- Saves the `course_feature_tags` section of OCW course parsed JSON into a new database field 
- Includes that field in the search index.
- Adds an aggregation for the tags
- Removes some code that's no longer needed

#### How should this be manually tested?
- If you have already run `manage.py backpopulate_ocw_data --overwrite` with a version of ocw-data-parser that saves a `course_feature_tags` section to the parsed JSON, then after running the migration in this PR your OCW courses should have a populated `course_feature_tags` field.
- If not, run `backpopulate_ocw_data` and interrupt after 10 minutes or so, or alternatively run `course_catalog.api.sync_ocw_courses`  in a shell on 1 or 2 courses, for example:
  ```sync_ocw_courses(course_prefixes=["PROD/6/6.730/Spring_2003/6-730-physics-for-solid-state-applications-spring-2003/"], blocklist=[], force_overwrite=True, upload_to_s3=False)```. When the import is complete, those courses should have a populated `course_feature_tags` field.
- Run `manage.py recreate_index`.  
- Search for certain courses that have some tags.  Check the network tab in your browser for the search API results and verify that the hits include `course_feature_tags` as a list, for example:
```
"course_feature_tags":["Problem Sets with Solutions","Exams with Solution","Lecture Notes"]
```
There should also be an aggregation for "course_feature_tags" in the response.


#### Any background context you want to provide?
- I decided to save the tags in a new database field rather than extract them from the `raw_json` field in the ES serializer during indexing, because that field is [deliberately excluded](https://github.com/mitodl/open-discussions/blob/3015b703ae8ae10249ce5b23f64a8bcb691b925a/search/serializers.py#L840-L842) to avoid memory issues.
